src/frontend/next/src/pages/_app.tsx

```diff
@@ -18,24 +20,44 @@ const App = ({ Component, pageProps }: AppProps) => {
     }
   }, []);
 
-  const [theme, setTheme] = useState(lightTheme);
+  const preferredTheme = usePreferredThemeColors();
+
+  const [theme, setTheme] = useState(preferredTheme);
+  const [mounted, setMounted] = useState(false);
+
+  // This hook is to ensure that right after the component mounted aka on client, display the client version of the app
+  useEffect(() => {
+    setMounted(true);
+  }, []);
 
   const toggleTheme = () => {
```

Comment:
Do this in your context code, where I'm commenting about that empty function, then you don't need to do this here, and you can expose your hook to get the current value at all times vs. using state.  See how I do the UserContext.

Reply:
@humphd : Hi Dave, I tried to find a way to accommodate what you mentioned in the comment but it's a bit challenging for me. My approach here is to follow the **Updating Context from a Nested Component** from the react page https://reactjs.org/docs/context.html#dynamic-context.

I'm not too familiar with context. Can you guide me through your comment?